src/controllers/auth.controller.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+const { ApiError } = require('../exceptions/errors');
+const { User } = require('../models/user');
+const userService = require('../services/user.service');
+const jwtService = require('../services/jwt.service');
+const tokenService = require('../services/token.service');
+const bcrypt = require('bcrypt');
+const { validateEmail,
+  validatePassword,
+  validateName } = require('../utils/validators');
+
+async function generateTokens(res, user) {
+  const normalizedUser = userService.normalize(user);
+  const accessToken = jwtService.sign(normalizedUser);
+  const refreshAccessToken = jwtService.signRefresh(normalizedUser);
+
+  await tokenService.save(normalizedUser.id, refreshAccessToken);
+
+  res.cookie('refreshToken', refreshAccessToken, {
+    HttpOnly: true,
+    maxAge: 30 * 24 * 60 * 60 * 1000,
+  });
+
+  res.send({
+    user: normalizedUser,
+    accessToken,
+  });
+}
+
+const register = async(req, res, next) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+    name: validateName(name),
+  };
```

Comment:
It's better to provide validations in middlewares, you can also use a npm package for this purpose